nameserver/nameserver_test.go

```diff
@@ -266,3 +332,192 @@ func TestTombstoneDeletion(t *testing.T) {
 	nameserver.deleteTombstones()
 	require.Equal(t, Entries{}, nameserver.entries)
 }
+
+// TestRestore tests whether entries have been restored after the nameserver restart
+// and after manually calling RestoreEntries.
+func TestRestore(t *testing.T) {
```

Comment:
Suggest that, instead of adding lengthy hard-coded test instances, `TestNameservers` is extended to include container ID and containers being stopped, restarted and destroyed.